### ISSUE TITLE
some setting question

### ISSUE THREAD

**Issue by JiangDong-miao** [MrPio/KAN-Continual_Learning_tests#1]
Hello, thank you so much for your sharing.
I read your code about the test of KAN, MLP.
But I have a small question. 
in the file 
https://github.com/MrPio/KAN-Continual_Learning_tests/blob/master/Is_EfficientKAN_good.ipynb
you init the KAN module from the efficientKAN, with the hyper-parameter

grid_size=200,
grid_eps=1,
scale_base=0,

which have a big difference with the defualt setting like 
grid_size=5,
scale_base=1.0,
grid_eps=0.02,

Is there any specific reason that you set like this?

**Comment by MrPio** [comment]
Hi! As you can read [here](https://github.com/Blealtan/efficient-kan/issues/38#issuecomment-2262860477), I have taken these hyper-parameters directly from the example in PyKAN 8, which is what the purpose of this test was to mimic.

**Comment by JiangDong-miao** [comment]
ahhh, OK, I got it , thank you so much.

By the way.. do you find some tricks about the choose of hyper-parameter in this case?

**Comment by JiangDong-miao** [comment]
and sorry, the dtype become float64 after passing the latent code to this module. 
would it effect the performance if not use the double() ?

**Comment by MrPio** [comment]
> ahhh, OK, I got it , thank you so much.
> 
> By the way.. do you find some tricks about the choose of hyper-parameter in this case?

Well, I only tried different combinations of learning rate and batch size, because the other hyperparameters have been copied from the PyKAN 8th example.

One thing that made me struggle a lot was the fact that I was using AdamW instead of the regular Adam optimiser. AdamW does the weight decay during the move as a form of regularisation, but in this continual learning scenario, this is just a detriment. Essentially, it lowered the old peaks while training the new ones.

However, the most important change I introduced concerns the `sp_trainable` and `sb_trainable` hyperparameters which were missing in EfficientKAN. To do this, I simply added two lines of code to the EfficientKAN layer class to disable training for the spline scaler and basis function scaler weights.

> and sorry, the dtype become float64 after passing the latent code to this module. would it effect the performance if not use the double() ?

No, I don't think so. I am quite confident I have run the test at least once without `self.double()` and seen no significant differences. I added that line because of some warning during the training.
